cpp/validate_harvested_records.cc

```diff
@@ -39,274 +40,195 @@
 namespace {
 
 
-[[noreturn]] void Usage() {
-   ::Usage("([--update-db-errors] marc_input marc_output missed_expectations_file email_address)|(update_db zeder_id zeder_instance field_name field_presence)\n"
-           "\tThis tool has two operating modes 1) checking MARC data for missed expectations and 2) altering these expectations.\n"
-           "\tin the \"update_db\" mode, \"field_name\" must be a 3-character MARC tag and \"field_presence\" must be one of\n"
-           "\tALWAYS, SOMETIMES, IGNORE.  Please note that only existing entries can be changed!"
-           "\tIf --update-db-errors is set, the errors_detected field will be reset for all journals of the found groups, and set only for the journals with at least 1 detected error.");
-}
-
-
 enum FieldPresence { ALWAYS, SOMETIMES, IGNORE };
 
 
-bool StringToFieldPresence(const std::string &field_presence_str, FieldPresence * const field_presence) {
-    if (field_presence_str == "ALWAYS") {
-        *field_presence = ALWAYS;
-        return true;
-    }
-
-    if (field_presence_str == "SOMETIMES") {
-        *field_presence = SOMETIMES;
-        return true;
-    }
-
-    if (field_presence_str == "IGNORE") {
-        *field_presence = IGNORE;
-        return true;
-    }
+FieldPresence StringToFieldPresence(const std::string &field_presence_str) {
+    if (field_presence_str == "ALWAYS")
+        return ALWAYS;
+    if (field_presence_str == "SOMETIMES")
+        return SOMETIMES;
+    if (field_presence_str == "IGNORE")
+        return IGNORE;
 
-    return false;
+    LOG_ERROR("unknown field presence \"" + field_presence_str + "\"!");
 }
 
 
-struct FieldInfo {
-    std::string name_;
-    FieldPresence presence_;
+class FieldRule {
```

Comment:
This looks like a container to me that may contain multiple rules, so it might be better to call the class `FieldRules` instead of `FieldRule`

Reply:
Done